azavea/loam
# Add TypeScript support and add types for GDALDataset

## Overview
Add TypeScript support and add types for GDALDataset to dogfood the changes.

### Notes
I think it's best to review this commit by commit since I detailed what I did for each step and why. I tried to make the minimal number of changes required to integrate TypeScript while still allowing the project to test and build as normal. I ended up upgrading lots of dev dependencies to make it happen, several of which were 2 major revisions behind, so we get upgraded dependencies as a bonus.

One note about tests is they will require rebuilding to take new changes to source files into account (this seems to have already been the case but I figured I'd call it out).

Another small thing to note is that as of webpack 4+ there is a `mode` argument that is either "developmen" or "production" which I believe corresponds to the arguments "dev" and "build", so we could potentially do away with those custom arguments, drop `yargs` as a dev dependency, and use the support for modifying config that comes with webpack: https://webpack.js.org/configuration/mode/ 

## Testing Instructions
* Review commands in `package.json` and run them all to make sure they all work
* Manually test Loam somehow to ensure it's still working and also that TypeScript source mapping is working (I wasn't sure how to test this; I think setting a breakpoint somewhere in `src/gdalDataset.ts` would work)

## Checklist


Resolves #373
